.github/actions/enforce-pr-conventions/action.yml

```diff
@@ -60,8 +63,22 @@ runs:
           echo "valid-pr-title-regex=$PR_REGEX" >> $GITHUB_OUTPUT
         fi
 
-    - name: Check branch name
+    - name: Check whitelisted branches
       if: steps.check.outputs.continue == 'true'
+      id: check-whitelist-branches
+      shell: bash
+      run: |
+        readarray -t branches <<< "${{ inputs.whitelist-branches }}"
+        for branch in "${branches[@]}"; do
+            if [[ $branch == $BRANCH_NAME ]]; then
+                echo "continue=false" >> $GITHUB_OUTPUT
+                exit 0
+            fi
+        done
+        echo "continue=true" >> $GITHUB_OUTPUT
```

Comment:
Would that work?

```suggestion
        if [[ ${branches[@]} =~ $BRANCH_NAME ]]; then
          echo "continue=false" >> $GITHUB_OUTPUT
        else
          echo "continue=true" >> $GITHUB_OUTPUT
        fi
```

Reply:
I guess it wont work because `=~` checks the regex pattern or maybe substring. I already tested this.
Eg if whitelisted-branch is `bug/AAE-123-conventional-commits` and if the current branch is `bug/AAE-123-conventional` then `if [[ ${branches[@]} =~ $BRANCH_NAME ]]` condition is true which is not correct.

Reply:
Alright thanks :+1: